framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -390,30 +390,39 @@ public S initialStore(UnderlyingAST underlyingAST, List<LocalVariableNode> param
     // the declared type instead of a refined type. Issue a warning to alert users?
     private void addInitialFieldValues(S store, ClassTree classTree, MethodTree methodTree) {
         boolean isConstructor = TreeUtils.isConstructor(methodTree);
+        boolean isStaticMethod =
+                ElementUtils.isStatic(TreeUtils.elementFromDeclaration(methodTree));
         TypeElement classEle = TreeUtils.elementFromDeclaration(classTree);
         for (FieldInitialValue<V> fieldInitialValue : analysis.getFieldInitialValues()) {
             VariableElement varEle = fieldInitialValue.fieldDecl.getField();
+            boolean fieldNotFromEnclosingClass = varEle.getEnclosingElement().equals(classEle);
```

Comment:
I would also move this variable after the check for static method and field, as it won't be used in that case.

Reply:
Okay, I move it after the first `if` clause